frontend/dynamic-demo-plugin/src/utils/telemetry.ts

```diff
@@ -0,0 +1,89 @@
+const SEGMENT_KEY = '';
+
+const initSegment = () => {
+  const analytics = ((window as any).analytics = (window as any).analytics || []);
+  if (!analytics.initialize) {
```

Comment:
Prefer early returns, which are easier to understand.

```suggestion
  if (analytics.initialize) {
    return;
  }
  // ...rest of logic...
```